src/widgets/kernel/qtooltip.cpp

```diff
@@ -475,6 +534,82 @@ void QToolTip::showText(const QPoint &pos, const QString &text, QWidget *w, cons
     }
 }
 
+/*!
+    Shows \a layout as a tool tip, with the global position \a pos as
+    the point of interest. The tool tip will be shown with a platform
+    specific offset from this point of interest.
+
+    If you specify a non-empty rect the tip will be hidden as soon
+    as you move your cursor out of this area.
+
+    The \a rect is in the coordinates of the widget you specify with
+    \a w. If the \a rect is not empty you must specify a widget.
+    Otherwise this argument can be \nullptr but it is used to
+    determine the appropriate screen on multi-head systems.
+
+    The \a msecDisplayTime parameter specifies for how long the tool tip
+    will be displayed, in milliseconds. With the default value of -1, the
+    time is based on the count of the layout.
+
+    If \a layout is empty the tool tip is hidden. If the layout is the
+    same as the currently shown tooltip, the tip will \e not move.
+    You can force moving by first hiding the tip with an empty layout,
+    and then showing the new tip at the new position.
+*/
+
+void QToolTip::showLayout(const QPoint& pos, QLayout* layout, QWidget* w, const QRect& rect, int msecDisplayTime)
+{
+    if (!layout)
+        return;
```

Comment:
Should this not also hide the current tip if it exists?

Reply:
do you mean, to treat situation when layout is set to NULL as a way to hide the layout? If so, yes, it makes sense.

Reply:
Yeah - I think sip will convert python None to nullptr, so that would be one way to hide it from python

Reply:
Why would anyone call `showLayout` without a proper Layout provided? Shouldn't it be misuse and instead throw some sort of error or assert failure?

Reply:
It's to mirror `showText` - to quote from the docs of `hideText`: "Hides the tool tip. This is the same as calling showText() with an empty string."

Reply:
You could argue passing an empty layout for that, but that's quite a different thing to an empty string, so nullptr also makes sense

Reply:
So after thinking about it for a little bit more, I think it needs further work. 
1. Something should destroy layout, sub layouts and widgets inside layouts when I'm hiding it.
2. `showText` should be also updated and check for layout and do the same as I described above. So that in a rare case when there a transition from showing a layout to showing text we can destroy the layout. 
3. Since I need to add further changes, would be good to define `hideLayout` method too.

Reply:
Whoops, you're right 🌴